### DISCUSSION TITLE
Adding images to pdf from memory

### DISCUSSION THREAD

**Discussion by BlueDiamondHead** [py-pdf/fpdf2#303] [Q&A]
Hi everyone,
I was wondering if it is possible to add images to a pdf where the image is a SVG which is code generated. I'm using the pdf.image function but it requires an image path. This is the first time I'm using python for a project so any comments are most welcome.
````
from fpdf import FPDF
import chess
import chess.svg

FEN = "8/8/8/8/4N3/8/8/8 w - - 0 1"

board = chess.Board(FEN)
boardsvg = chess.svg.board(board,size=600)

f = open("BoardVisualisedFromFEN.SVG", "w")
f.write(boardsvg)
f.close()

from svglib.svglib import svg2rlg
from reportlab.graphics import renderPM

image_path = "BoardVisualisedFromFEN.SVG"
drawing = svg2rlg(image_path)
renderPM.drawToFile(drawing,"file.png",fmt='PNG')

Width = 210
Height = 297
Margin = 25
ImageMargin = Margin/2

NumberVerticalImages = 3
NumberHorizontalImages = 2

ImageHeigth = (Height-2*Margin-ImageMargin*(NumberVerticalImages-1))/NumberVerticalImages
ImageWidth = (Width - 2*Margin-ImageMargin*(NumberHorizontalImages-1))/NumberHorizontalImages

ImageSize = min(ImageHeigth,ImageWidth)

pdf = FPDF()
pdf.add_page()
pdf.image("file.png",Margin,Margin,w=ImageSize,h=ImageSize)`
```

**Comment by Lucas-C** [comment]
Hi @BlueDiamondHead

[FPDF.image](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.image) accepts an instance of `PIL.Image.Image` as argument.

For SVG support, check out https://github.com/PyFPDF/fpdf2/issues/19 & https://github.com/PyFPDF/fpdf2/pull/196.

**Reply by Lucas-C** [reply]
FYI @BlueDiamondHead SVG support has been added to `fpdf2` thanks to @torque!
Doc: https://pyfpdf.github.io/fpdf2/SVG.html
It has been released in v[2.5.0](https://github.com/PyFPDF/fpdf2/blob/master/CHANGELOG.md#250---2022-01-22)

The feature to pass `.svg` images from memory (as `BytesIO`) to `FPDF.image` has just been merged: #337

You can test it already by installing `fpdf2` directly from the `master` GitHub branch:

    pip install git+https://github.com/PyFPDF/fpdf2.git@master
